nixos/modules/services/search/solr.nix

```diff
@@ -6,142 +6,107 @@ let
 
   cfg = config.services.solr;
 
-  # Assemble all jars needed for solr
-  solrJars = pkgs.stdenv.mkDerivation {
-    name = "solr-jars";
-
-    src = pkgs.fetchurl {
-      url = http://archive.apache.org/dist/tomcat/tomcat-5/v5.5.36/bin/apache-tomcat-5.5.36.tar.gz;
-      sha256 = "01mzvh53wrs1p2ym765jwd00gl6kn8f9k3nhdrnhdqr8dhimfb2p";
-    };
-
-    installPhase = ''
-      mkdir -p $out/lib
-      cp common/lib/*.jar $out/lib/
-      ln -s ${pkgs.ant}/lib/ant/lib/ant.jar $out/lib/
-      ln -s ${cfg.solrPackage}/lib/ext/* $out/lib/
-      ln -s ${pkgs.jdk.home}/lib/tools.jar $out/lib/
-    '' + optionalString (cfg.extraJars != []) ''
-      for f in ${concatStringsSep " " cfg.extraJars}; do
-         cp $f $out/lib
-      done
-    '';
-  };
-
-in {
+in
 
+{
   options = {
     services.solr = {
       enable = mkOption {
```

Comment:
shoudln't this be using `mkEnable` instead?

Reply:
I've never seen a service have .enable as mkEnable instead of mkOption. A quick search on github didn't show any instances either. Could you please elaborate possibly with a link to an example?

Reply:
Sorry, I meant `mkEnableOption` it just takes one argument (the description) so that en tire enable block would just become:


```nix
{
  enable = mkEnableOption "Enables the solr service.";
}
```

Reply:
Ah cool I hadn't noticed that one before. Sounds good.